lib/qesapdeployment.pm

```diff
@@ -486,6 +487,73 @@ sub qesap_execute {
     return @results;
 }
 
+=head3 qesap_execute_conditional_retry
+
+    qesap_execute(cmd => $qesap_script_cmd [, verbose => 1, cmd_options => $cmd_options] );
+    cmd_options - allows to append additional qesap.py commands arguments
+    like "qesap.py terraform -d"
+        Example:
+        qesap_execute(cmd => 'terraform', cmd_options => '-d') will result in:
+        qesap.py terraform -d
+
+    Execute qesap glue script commands. Check project documentation for available options:
+    https://github.com/SUSE/qe-sap-deployment
+    Test only returns execution result, failure has to be handled by calling method.
+
+=over 7
+
+=item B<CMD> - qesap.py subcommand to run
+
+=item B<CMD_OPTIONS> - set of arguments for the qesap.py subcommand
+
+=item B<VERBOSE> - activate verbosity in qesap.py
+
+=item B<TIMEOUT> - max expected execution time
+
+=item B<LOGNAME> - filename of the log file. This argument is optional,
+                   if not specified the log filename is internally calculated
+                   using content from CMD and CMD_OPTIONS.
+
+=item B<RETRIES> - number of retries in case of expected error
+
+=item B<ERROR_STRING> - error string to look for
+
+=back
+=cut
+
+sub qesap_execute_conditional_retry {
+    my (%args) = @_;
+    croak 'Missing mandatory cmd argument' unless $args{cmd};
+    croak 'Missing mandatory error string' unless $args{error_string};
+    my $verbose = $args{verbose} ? "--verbose" : "";
+    $args{cmd_options} ||= '';
+    $args{timeout} //= bmwqemu::scale_timeout(90);
+    $args{logname} //= '';
+    $args{retries} //= 1;
```

Comment:
Does this argument have an effect? I am not able to find the line which considers it.

Reply:
Fixed! Now the code actually uses the retries arg